ansible_runner/receptor_plugin.py

```diff
@@ -0,0 +1,162 @@
+import logging
+import json
+import os
+import time
+import io
+import tarfile
+import tempfile
+import uuid
+import asyncio
+
+try:
+    import receptor
+    from receptor.config import ReceptorConfig
+    from receptor.controller import Controller
+    receptor_import = True
+except ImportError:
+    receptor_import = False
+
+
+from ansible_runner import run
+
+
+logger = logging.getLogger(__name__)
+
+
+class UUIDEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, uuid.UUID):
+            return obj.hex
+        return json.JSONEncoder.default(self, obj)
+
+
+# List of kwargs options to the run method that should be sent to the remote executor.
+remote_run_options = (
+    'forks',
+    'host_pattern',
+    'ident',
+    'ignore_logging',
+    'inventory',
+    'limit',
+    'module',
+    'module_args',
+    'omit_event_data',
+    'only_failed_event_data',
+    'playbook',
+    'verbosity',
+)
+
+
+def run_via_receptor(receptor_node, receptor_peer, receptor_node_id, run_options):
+
+    async def read_responses():
+        while True:
+            message = await controller.recv()
```

Comment:
If a network blip happens and receptor disconnects. What happens?

Reply:
Receptor has internal reconnect logic, and won't drop buffer content on disconnect.  It's supposed to be resilient to transient network errors.

For the temporary receptor node inside Runner, it won't reconnect and will error out.  This is kind of unavoidable with the ephemeral node architecture and is one of the motivations for switching to a Unix domain socket instead of ephemeral Receptor-protocol nodes.